sklearn/utils/extmath.py

```diff
@@ -1000,39 +1008,6 @@ def make_nonnegative(X, min_value=0):
     return X
 
 
-# Use at least float64 for the accumulating functions to avoid precision issue
-# see https://github.com/numpy/numpy/issues/9393. The float64 is also retained
-# as it is in case the float overflows
-def _safe_accumulator_op(op, x, *args, **kwargs):
-    """
-    This function provides numpy accumulator functions with a float64 dtype
-    when used on a floating point input. This prevents accumulator overflow on
-    smaller floating point dtypes.
-
-    Parameters
-    ----------
-    op : function
-        A numpy accumulator function such as np.mean or np.sum.
-    x : ndarray
-        A numpy array to apply the accumulator function.
-    *args : positional arguments
-        Positional arguments passed to the accumulator function after the
-        input x.
-    **kwargs : keyword arguments
-        Keyword arguments passed to the accumulator function.
-
-    Returns
-    -------
-    result
-        The output of the accumulator function passed to this function.
-    """
-    if np.issubdtype(x.dtype, np.floating) and x.dtype.itemsize < 8:
-        result = op(x, *args, **kwargs, dtype=np.float64)
-    else:
-        result = op(x, *args, **kwargs)
-    return result
-
-
```

Comment:
why the move of `_safe_accumulator_op` in `_array_api`? I think we should keep those helpers (even once array api-ified) in `utils.extmath` and only add to `_array_api` what we anticipate could be part of a future version of the Array API standard.

Reply:
(diff has been reverted since)